### DISCUSSION TITLE
A way of storing set of objects in posix mode.

### DISCUSSION THREAD

**Discussion by Irbis03** [versity/versitygw#865] [Ideas]
Excuse me, please tell me, are there plans in the future to add the ability to store a set of objects in a single "volume" in posix mode?

The reason why I'm asking this: We have a distributed posix compatible file system acting as storage for versirtygw, and storing a large number of small objects in a "volume" would allow for more efficient use of file system resources.

**Comment by benmcclelland** [comment]
Just to clarify, "volume" in this case would be a single file within the filesystem? There are currently no plans to change the direct mapping of object directly to single file. Generally to combine files into a single container, you would need to do this as a batch operation. So either on the client side before uploading, or some background task within the filesystem that looks for candidate files to combine into containers. Inline with S3 is not a very good candidate for this since each request is an individual object. So there is no opportunity within the request to combine with other incoming objects.

**Reply by Irbis03** [reply]
I understood, thank you!

**Reply by scaleoutsean** [reply]
> Generally to combine files into a single container, you would need to do this as a batch operation. 

I think the classic cheat for PFS is to create a blob and loopback-mount it. Then run VGW on that mount. Objects are stored on the FS but without going through S3 GW you'd see a large "container" blob. Still, for the purpose of S3 on a PFS that can't tolerate many small files this should work and it's probably even better (that users can't see objects/files in POSIX mode).
